index/scorch/introducer.go

```diff
@@ -230,3 +241,50 @@ func (s *Scorch) introduceMerge(nextMerge *segmentMerge) {
 	// notify merger we incorporated this
 	close(nextMerge.notify)
 }
+
+func (s *Scorch) revertToSnapshot(revertTo *snapshotRevert) error {
+	// acquire lock
+	s.rootLock.Lock()
+
+	// prepare a new index snapshot, based on next snapshot
+	newSnapshot := &IndexSnapshot{
+		parent:   s,
+		segment:  make([]*SegmentSnapshot, len(revertTo.snapshot.segment)),
+		offsets:  make([]uint64, len(revertTo.snapshot.offsets)),
```

Comment:
as an optimization - i suspect you don't  actually need to copy the offsets here?  or copy the internal?  can just share the same instance, and let GC do all the hard work of releasing memory if unused?

Reply:
Yeah I guess, I'll change this in the next patch.